core/linux/test/TimeZoneNativeTest.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019-2025 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.test
+
+import kotlinx.datetime.IllegalTimeZoneException
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.internal.root
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+class TimeZoneNativeTest {
+
+    @Test
+    fun correctSymlinkTest() = withFakeRoot("${RESOURCES}correct-symlink/") {
+        val tz = TimeZone.currentSystemDefault()
+        assertEquals(TimeZone.of("Europe/Oslo"), tz)
+    }
+
+    @Test
+    fun correctLocaltimeCopyTest() = withFakeRoot("${RESOURCES}correct-localtime-copy/") {
+        val tz = TimeZone.currentSystemDefault()
+        assertEquals(TimeZone.of("Europe/Oslo"), tz)
+    }
+
+    @Test
+    fun fallsBackToUTC() = withFakeRoot("${RESOURCES}falls-back-to-utc/") {
+        val tz = TimeZone.currentSystemDefault()
+        assertEquals(TimeZone.UTC, tz)
+    }
+
+    @Test
+    fun missingTimezoneTest() = withFakeRoot("${RESOURCES}missing-timezone/") {
```

Comment:
Ditto w.r.t. clearly describing what the test checks: it's normal that `/etc/timezone` doesn't exist, so what this test checks is what happens if `/etc/localtime` is a not a symlink, *but* the `/etc/timezone` file doesn't exist.